UnitsNet/CustomCode/UnitParser.cs

```diff
@@ -54,11 +55,27 @@ internal static class UnitParser
 
             const string exponentialRegex = @"(?:[eE][-+]?\d+)?)";
 
+            // Special regex characters that need escaping
+            string[] regexSpecialCharacters = {"$","^","*","?"};
+            string[] unitAbbreviations = UnitSystem.GetCached(formatProvider)
+                .GetAllAbbreviations(unitType)
+                .OrderByDescending(s => s.Length) // Important to order by length -- if "m" is before "mm" and the input is "mm", it will match just "m" and throw invalid string error
+                .ToArray();
+            // Escape special regex characters
+            for (int i = 0; i < unitAbbreviations.Length; i++)
```

Comment:
That looks like a much better approach. By the looks of it, I will still need a for loop though? If the project is ever updated to .NET Standard 2.0, the for loop can be replaced by:
`string unitsRegex = $"({String.Join("|", unitAbbreviations.ForEach(s => Regex.Replace(s)))})";`